File: CS/Grid MVVM Filtering/ViewModel/Common/ObservableObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.ComponentModel;

namespace Grid_MVVM_Filtering
{
    public abstract class ObservableObject : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged;

        protected void OnPropertyChanged(string propertyName)
        {
            if (PropertyChanged != null)
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
        }
    }
}

File: VB/Grid MVVM Filtering/ViewModel/Common/ObservableObject.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.ComponentModel

Namespace Grid_MVVM_Filtering
	Public MustInherit Class ObservableObject
		Implements INotifyPropertyChanged
		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

		Protected Sub OnPropertyChanged(ByVal propertyName As String)
			RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
		End Sub
	End Class
End Namespace